gateway/handlers/infohandler.go

```diff
@@ -0,0 +1,56 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/openfaas/faas/gateway/types"
+	"github.com/openfaas/faas/gateway/version"
+)
+
+// MakeInfoHandler is responsible for display component version information
+func MakeInfoHandler(h http.Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		sw := types.NewStringResponseWriter()
```

Comment:
This looks interesting. I've used a http recorder elsewhere for this task such `add_metrics.go`

Reply:
Good idea